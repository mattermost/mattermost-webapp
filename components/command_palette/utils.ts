// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Channel} from '@mattermost/types/channels';
import {Team} from '@mattermost/types/teams';

// See LICENSE.txt for license information.
import {CommandPaletteItem} from './command_palette_list_item/command_palette_list_item';
import {
    CmdPalettePictographIcon,
    CmdPalettePictographType,
} from './command_palette_list_item/command_palette_list_item_pictograph';
import {ChannelType, CommandPaletteEntities} from './types';

const channelTypeOrder = {
    [ChannelType.OPEN_CHANNEL]: 0,
    [ChannelType.PRIVATE_CHANNEL]: 1,
    [ChannelType.DM_CHANNEL]: 2,
    [ChannelType.GM_CHANNEL]: 3,
};

function filterName(name: string): string {
    return name.replace(/[.,'"\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // eslint-disable-line no-useless-escape
}

export function channelToCommandPaletteItemTransformer(channels: Channel[], teams: Record<string, Team>): CommandPaletteItem[] {
    return channels.map((channel) => {
        const isArchived = channel.delete_at ? channel.delete_at !== 0 : false;

        let channelIcon = CmdPalettePictographIcon.GLOBE;
        if (isArchived) {
            channelIcon = CmdPalettePictographIcon.ARCHIVE;
        }
        return {
            description: channel.type === ChannelType.DM_CHANNEL ? channel.name : '',
            id: channel.id,
            isDeactivated: isArchived,
            teamName: teams[channel.team_id]?.name,
            pictograph: {
                type: CmdPalettePictographType.ICON,
                pictographItem: channelIcon,
            },
            subType: channel.type as ChannelType,
            title: channel.display_name,
            type: CommandPaletteEntities.Channel,
            isArchived,
        };
    });
}

function sortChannelsByTypeAndDisplayName(locale: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem): number {
    if (channelTypeOrder[cmdPaletteItemA.subType] !== channelTypeOrder[cmdPaletteItemB.subType]) {
        if (channelTypeOrder[cmdPaletteItemA.subType] < channelTypeOrder[cmdPaletteItemB.subType]) {
            return -1;
        }

        return 1;
    }

    const aDisplayName = filterName(cmdPaletteItemA.title);
    const bDisplayName = filterName(cmdPaletteItemB.title);

    if (aDisplayName !== bDisplayName) {
        return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
    }

    return cmdPaletteItemA.title.toLowerCase().localeCompare(cmdPaletteItemA.title.toLowerCase(), locale, {numeric: true});
}

function sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.lastViewedAt && cmdPaletteItemB.lastViewedAt) {
        return cmdPaletteItemB.lastViewedAt - cmdPaletteItemA.lastViewedAt;
    } else if (cmdPaletteItemA.lastViewedAt) {
        return -1;
    } else if (cmdPaletteItemB.lastViewedAt) {
        return 1;
    }

    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
    return sortChannelsByTypeAndDisplayName('en', cmdPaletteItemA, cmdPaletteItemB);
}

export function commandPaletteSorter(prefix: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.isArchived && !cmdPaletteItemB.isArchived) {
        return 1;
    } else if (!cmdPaletteItemA.isArchived && cmdPaletteItemB.isArchived) {
        return -1;
    }

    if (cmdPaletteItemA.isDeactivated && !cmdPaletteItemB.isDeactivated) {
        return 1;
    } else if (cmdPaletteItemB.isDeactivated && !cmdPaletteItemA.isDeactivated) {
        return -1;
    }

    let aDisplayName = cmdPaletteItemA.title.toLowerCase();
    let bDisplayName = cmdPaletteItemA.title.toLowerCase();

    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.DM_CHANNEL &&
        aDisplayName.startsWith('@')) {
        aDisplayName = aDisplayName.substring(1);
    }

    if (cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.DM_CHANNEL && bDisplayName.startsWith('@')) {
        bDisplayName = bDisplayName.substring(1);
    }

    const aStartsWith = aDisplayName.startsWith(prefix) || cmdPaletteItemA.description.toLowerCase().startsWith(prefix);
    const bStartsWith = bDisplayName.startsWith(prefix) || cmdPaletteItemB.description.toLowerCase().startsWith(prefix);

    // Open channels user haven't interacted should be at the  bottom of the list
    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.OPEN_CHANNEL &&
        !cmdPaletteItemA.lastViewedAt &&
        ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
            cmdPaletteItemB.subType !== ChannelType.OPEN_CHANNEL) || cmdPaletteItemB.lastViewedAt)) {
        return 1;
    } else if ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.OPEN_CHANNEL) || !cmdPaletteItemB.lastViewedAt) {
        return -1;
    }

    // Sort channels starting with the search term first
    if (aStartsWith && !bStartsWith) {
        return -1;
    } else if (!aStartsWith && bStartsWith) {
        return 1;
    }
    return sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA, cmdPaletteItemB);
}
