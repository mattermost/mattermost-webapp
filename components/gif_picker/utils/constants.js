// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    ItemTapAction: {
        OPEN_EMBED_PAGE: 1,
        SHARE: 2,
    },

    Tab: {
        TRENDING: 0,
        REACTIONS: 1,
    },

    appName: {
        mattermost: 'mattermost',
    },
};
