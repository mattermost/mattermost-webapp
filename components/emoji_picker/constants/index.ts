// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory} from '@mattermost/types/emojis';

import * as Emoji from 'utils/emoji';

import {Category, Categories} from '../types';

export const RECENT = 'recent';
export const SEARCH_RESULTS = 'searchResults';
export const SMILEY_EMOTION = 'smileys-emotion';
export const CUSTOM = 'custom';

const categoryClass: Map<EmojiCategory, string> = new Map([
    [RECENT, 'icon-clock-outline'],
    [SMILEY_EMOTION, 'icon-emoticon-happy-outline'],
    ['people-body', 'icon-account-outline'],
    ['animals-nature', 'icon-leaf-outline'],
    ['food-drink', 'icon-food-apple'],
    ['activities', 'icon-basketball'],
    ['travel-places', 'icon-airplane-variant'],
    ['objects', 'icon-lightbulb-outline'],
    ['symbols', 'icon-heart-outline'],
    ['flags', 'icon-flag-outline'],
    [CUSTOM, 'icon-emoticon-custom-outline'],
    [SEARCH_RESULTS, ''],
]);

function createCategory(name: EmojiCategory): Category {
    return {
        name,
        id: Emoji.CategoryTranslations.get(name) || '',
        className: categoryClass.get(name) || '',
        message: Emoji.CategoryMessage.get(name)!,
    };
}

export const RECENT_EMOJI_CATEGORY: Pick<Categories, 'recent'> = {recent: createCategory(RECENT)};
export const SEARCH_EMOJI_CATEGORY: Pick<Categories, typeof SEARCH_RESULTS> = {searchResults: createCategory(SEARCH_RESULTS)};

export const CATEGORIES: Categories = Emoji.CategoryNames.
    filter((category) => !(category === 'recent' || category === 'searchResults')).
    reduce((previousCategory, currentCategory) => {
        return {
            ...previousCategory,
            [currentCategory]: createCategory(currentCategory as EmojiCategory),
        };
    }, {} as Categories);

export const EMOJI_PER_ROW = 9; // needs to match variable `$emoji-per-row` in _variables.scss
export const ITEM_HEIGHT = 36; //as per .emoji-picker__item height in _emoticons.scss
export const EMOJI_CONTAINER_HEIGHT = 290; // If this changes, the spaceRequiredAbove and spaceRequiredBelow props passed to the EmojiPickerOverlay must be updated

export const CATEGORY_HEADER_ROW = 'categoryHeaderRow';
export const EMOJIS_ROW = 'emojisRow';

export const EMOJI_SCROLL_THROTTLE_DELAY = 150;
export const EMOJI_ROWS_OVERSCAN_COUNT = 1;

export const CUSTOM_EMOJIS_PER_PAGE = 200;
export const CUSTOM_EMOJI_SEARCH_THROTTLE_TIME_MS = 1000;
