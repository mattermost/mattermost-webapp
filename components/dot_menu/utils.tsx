// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {trackEvent} from 'actions/telemetry_actions';

import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
