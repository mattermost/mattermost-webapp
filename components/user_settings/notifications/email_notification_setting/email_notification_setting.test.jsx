// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Preferences} from 'utils/constants';

import {mountWithIntl} from 'tests/helpers/intl-test-helper.jsx';
import EmailNotificationSetting from 'components/user_settings/notifications/email_notification_setting/email_notification_setting.jsx';

describe('components/user_settings/notifications/EmailNotificationSetting', () => {
    const requiredProps = {
        currentUserId: 'current_user_id',
        activeSection: 'email',
        updateSection: jest.fn(),
        enableEmail: false,
        emailInterval: Preferences.INTERVAL_NEVER,
        onSubmit: jest.fn(),
        onCancel: jest.fn(),
        onChange: jest.fn(),
        serverError: '',
        saving: false,
        sendEmailNotifications: true,
        enableEmailBatching: false,
        siteName: 'Mattermost',
        actions: {
            savePreferences: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = mountWithIntl(<EmailNotificationSetting {...requiredProps}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('#emailNotificationImmediately').exists()).toBe(true);
        expect(wrapper.find('#emailNotificationNever').exists()).toBe(true);
        expect(wrapper.find('#emailNotificationMinutes').exists()).toBe(false);
        expect(wrapper.find('#emailNotificationHour').exists()).toBe(false);
    });

    test('should match snapshot, enabled email batching', () => {
        const props = {
            ...requiredProps,
            enableEmailBatching: true,
        };
        const wrapper = mountWithIntl(<EmailNotificationSetting {...props}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('#emailNotificationMinutes').exists()).toBe(true);
        expect(wrapper.find('#emailNotificationHour').exists()).toBe(true);
    });

    test('should match snapshot, not send email notifications', () => {
        const props = {
            ...requiredProps,
            sendEmailNotifications: false,
        };
        const wrapper = shallow(<EmailNotificationSetting {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, active section != email and SendEmailNotifications !== true', () => {
        const props = {
            ...requiredProps,
            sendEmailNotifications: false,
            activeSection: '',
        };
        const wrapper = shallow(<EmailNotificationSetting {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, active section != email and SendEmailNotifications = true', () => {
        const props = {
            ...requiredProps,
            sendEmailNotifications: true,
            activeSection: '',
        };
        const wrapper = shallow(<EmailNotificationSetting {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, active section != email, SendEmailNotifications = true and enableEmail = true', () => {
        const props = {
            ...requiredProps,
            sendEmailNotifications: true,
            activeSection: '',
            enableEmail: true,
        };
        const wrapper = shallow(<EmailNotificationSetting {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on serverError', () => {
        const newServerError = 'serverError';
        const props = {...requiredProps, serverError: newServerError};
        const wrapper = shallow(<EmailNotificationSetting {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should pass handleChange', () => {
        const wrapper = mountWithIntl(<EmailNotificationSetting {...requiredProps}/>);
        wrapper.find('#emailNotificationImmediately').simulate('change');

        expect(wrapper.state('enableEmail')).toBe('true');
        expect(wrapper.state('newInterval')).toBe(Preferences.INTERVAL_IMMEDIATE);
        expect(requiredProps.onChange).toBeCalledTimes(1);
    });

    test('should pass handleSubmit', async () => {
        const newOnSubmit = jest.fn();
        const newUpdateSection = jest.fn();
        const newSavePreference = jest.fn();
        const props = {
            ...requiredProps,
            onSubmit: newOnSubmit,
            updateSection: newUpdateSection,
            actions: {savePreferences: newSavePreference},
        };

        const wrapper = mountWithIntl(<EmailNotificationSetting {...props}/>);

        await wrapper.instance().handleSubmit();
        expect(wrapper.state('newInterval')).toBe(Preferences.INTERVAL_NEVER);
        expect(newOnSubmit).not.toBeCalled();
        expect(newUpdateSection).toHaveBeenCalledTimes(1);
        expect(newUpdateSection).toBeCalledWith('');

        const newInterval = Preferences.INTERVAL_IMMEDIATE;
        wrapper.find('#emailNotificationImmediately').simulate('change');
        await wrapper.instance().handleSubmit();

        expect(wrapper.state('newInterval')).toBe(newInterval);
        expect(newOnSubmit).toBeCalled();
        expect(newOnSubmit).toHaveBeenCalledTimes(1);

        const expectedPref = [{
            category: 'notifications',
            name: 'email_interval',
            user_id: 'current_user_id',
            value: newInterval.toString(),
        }];

        expect(newSavePreference).toHaveBeenCalledTimes(1);
        expect(newSavePreference).toBeCalledWith('current_user_id', expectedPref);
    });

    test('should pass handleUpdateSection', () => {
        const newUpdateSection = jest.fn();
        const newOnCancel = jest.fn();
        const props = {...requiredProps, updateSection: newUpdateSection, onCancel: newOnCancel};
        const wrapper = mountWithIntl(<EmailNotificationSetting {...props}/>);

        wrapper.instance().handleUpdateSection('email');
        expect(newUpdateSection).toBeCalledWith('email');
        expect(newUpdateSection).toHaveBeenCalledTimes(1);
        expect(newOnCancel).not.toBeCalled();

        wrapper.instance().handleUpdateSection();
        expect(newUpdateSection).toBeCalled();
        expect(newUpdateSection).toHaveBeenCalledTimes(2);
        expect(newUpdateSection).toBeCalledWith('');
        expect(newOnCancel).toBeCalled();
    });

    test('should derived state from props', () => {
        const nextProps = {
            emailInterval: Preferences.INTERVAL_IMMEDIATE,
            enableEmailBatching: true,
            sendEmailNotifications: true,
        };
        const wrapper = mountWithIntl(<EmailNotificationSetting {...requiredProps}/>);
        expect(wrapper.state('emailInterval')).toBe(requiredProps.emailInterval);
        expect(wrapper.state('enableEmailBatching')).toBe(requiredProps.enableEmailBatching);
        expect(wrapper.state('sendEmailNotifications')).toBe(requiredProps.sendEmailNotifications);
        expect(wrapper.state('newInterval')).toBe(Preferences.INTERVAL_NEVER);

        wrapper.setProps(nextProps);
        expect(wrapper.state('emailInterval')).toBe(nextProps.emailInterval);
        expect(wrapper.state('enableEmailBatching')).toBe(nextProps.enableEmailBatching);
        expect(wrapper.state('sendEmailNotifications')).toBe(nextProps.sendEmailNotifications);
        expect(wrapper.state('newInterval')).toBe(Preferences.INTERVAL_NEVER);

        nextProps.enableEmail = true;
        nextProps.emailInterval = Preferences.INTERVAL_FIFTEEN_MINUTES;
        wrapper.setProps(nextProps);
        expect(wrapper.state('emailInterval')).toBe(nextProps.emailInterval);
        expect(wrapper.state('enableEmailBatching')).toBe(nextProps.enableEmailBatching);
        expect(wrapper.state('sendEmailNotifications')).toBe(nextProps.sendEmailNotifications);
        expect(wrapper.state('newInterval')).toBe(Preferences.INTERVAL_FIFTEEN_MINUTES);
    });
});
