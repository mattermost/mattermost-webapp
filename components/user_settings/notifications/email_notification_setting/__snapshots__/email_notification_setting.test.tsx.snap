// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot 1`] = `
<EmailNotificationSetting
  actions={
    Object {
      "savePreferences": [MockFunction],
    }
  }
  activeSection="email"
  currentUserId="current_user_id"
  emailInterval={0}
  enableEmail="false"
  enableEmailBatching={false}
  onCancel={[MockFunction]}
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
  saving={false}
  sendEmailNotifications={true}
  serverError=""
  updateSection={[MockFunction]}
>
  <SettingItemMax
    containerStyle=""
    infoPosition="bottom"
    inputs={
      Array [
        <fieldset>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send email notifications"
              id="user.settings.notifications.email.send"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                data-email-interval={30}
                data-enable-email="true"
                id="emailNotificationImmediately"
                name="emailNotifications"
                onChange={[Function]}
                type="radio"
              />
              <FormattedMessage
                defaultMessage="Immediately"
                id="user.settings.notifications.email.immediately"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                data-email-interval={0}
                data-enable-email="false"
                id="emailNotificationNever"
                name="emailNotifications"
                onChange={[Function]}
                type="radio"
              />
              <FormattedMessage
                defaultMessage="Never"
                id="user.settings.notifications.email.never"
              />
            </label>
          </div>
          <div
            className="mt-3"
          >
            <FormattedMessage
              defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes."
              id="user.settings.notifications.emailInfo"
            />
             
          </div>
        </fieldset>,
      ]
    }
    saving={false}
    section=""
    server_error=""
    submit={[Function]}
    title="Email Notifications"
    updateSection={[Function]}
  >
    <section
      className="section-max form-horizontal "
    >
      <h4
        className="col-sm-12 section-title"
        id="settingTitle"
      >
        Email Notifications
      </h4>
      <div
        className="col-sm-9 col-sm-offset-3"
      >
        <div
          className="setting-list"
          tabIndex="-1"
        >
          <div
            className="setting-list-item"
          >
            <fieldset
              key="userNotificationEmailOptions"
            >
              <legend
                className="form-legend"
              >
                <FormattedMessage
                  defaultMessage="Send email notifications"
                  id="user.settings.notifications.email.send"
                >
                  <span>
                    Send email notifications
                  </span>
                </FormattedMessage>
              </legend>
              <div
                className="radio"
              >
                <label>
                  <input
                    checked={false}
                    data-email-interval={30}
                    data-enable-email="true"
                    id="emailNotificationImmediately"
                    name="emailNotifications"
                    onChange={[Function]}
                    type="radio"
                  />
                  <FormattedMessage
                    defaultMessage="Immediately"
                    id="user.settings.notifications.email.immediately"
                  >
                    <span>
                      Immediately
                    </span>
                  </FormattedMessage>
                </label>
              </div>
              <div
                className="radio"
              >
                <label>
                  <input
                    checked={true}
                    data-email-interval={0}
                    data-enable-email="false"
                    id="emailNotificationNever"
                    name="emailNotifications"
                    onChange={[Function]}
                    type="radio"
                  />
                  <FormattedMessage
                    defaultMessage="Never"
                    id="user.settings.notifications.email.never"
                  >
                    <span>
                      Never
                    </span>
                  </FormattedMessage>
                </label>
              </div>
              <div
                className="mt-3"
              >
                <FormattedMessage
                  defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes."
                  id="user.settings.notifications.emailInfo"
                >
                  <span>
                    Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes.
                  </span>
                </FormattedMessage>
                 
              </div>
            </fieldset>
          </div>
          <div
            className="setting-list-item"
          >
            <hr />
            <SaveButton
              btnClass="btn-primary"
              defaultMessage={
                <FormattedMessage
                  defaultMessage="Save"
                  id="save_button.save"
                />
              }
              disabled={false}
              extraClasses=""
              onClick={[Function]}
              saving={false}
              savingMessage={
                <FormattedMessage
                  defaultMessage="Saving"
                  id="save_button.saving"
                />
              }
            >
              <button
                className="save-button btn btn-primary"
                data-testid="saveSetting"
                disabled={false}
                id="saveSetting"
                onClick={[Function]}
                type="submit"
              >
                <LoadingWrapper
                  loading={false}
                  text={
                    <FormattedMessage
                      defaultMessage="Saving"
                      id="save_button.saving"
                    />
                  }
                >
                  <span>
                    <FormattedMessage
                      defaultMessage="Save"
                      id="save_button.save"
                    >
                      <span>
                        Save
                      </span>
                    </FormattedMessage>
                  </span>
                </LoadingWrapper>
              </button>
            </SaveButton>
            <button
              className="btn btn-sm btn-cancel cursor--pointer style--none"
              id="cancelSetting"
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Cancel"
                id="setting_item_max.cancel"
              >
                <span>
                  Cancel
                </span>
              </FormattedMessage>
            </button>
          </div>
        </div>
      </div>
    </section>
  </SettingItemMax>
</EmailNotificationSetting>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, active section != email and SendEmailNotifications !== true 1`] = `
<SettingItemMin
  describe={
    <FormattedMessage
      defaultMessage="Email notifications are not enabled"
      id="user.settings.notifications.email.disabled"
    />
  }
  section="email"
  title="Email Notifications"
  updateSection={[Function]}
/>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, active section != email and SendEmailNotifications = true 1`] = `
<SettingItemMin
  describe={
    <FormattedMessage
      defaultMessage="Never"
      id="user.settings.notifications.email.never"
    />
  }
  section="email"
  title="Email Notifications"
  updateSection={[Function]}
/>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, active section != email, SendEmailNotifications = true and enableEmail = true 1`] = `
<SettingItemMin
  describe={
    <FormattedMessage
      defaultMessage="Immediately"
      id="user.settings.notifications.email.immediately"
    />
  }
  section="email"
  title="Email Notifications"
  updateSection={[Function]}
/>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, enabled email batching 1`] = `
<EmailNotificationSetting
  actions={
    Object {
      "savePreferences": [MockFunction],
    }
  }
  activeSection="email"
  currentUserId="current_user_id"
  emailInterval={0}
  enableEmail="false"
  enableEmailBatching={true}
  onCancel={[MockFunction]}
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
  saving={false}
  sendEmailNotifications={true}
  serverError=""
  updateSection={[MockFunction]}
>
  <SettingItemMax
    containerStyle=""
    infoPosition="bottom"
    inputs={
      Array [
        <fieldset>
          <legend
            className="form-legend"
          >
            <FormattedMessage
              defaultMessage="Send email notifications"
              id="user.settings.notifications.email.send"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                data-email-interval={30}
                data-enable-email="true"
                id="emailNotificationImmediately"
                name="emailNotifications"
                onChange={[Function]}
                type="radio"
              />
              <FormattedMessage
                defaultMessage="Immediately"
                id="user.settings.notifications.email.immediately"
              />
            </label>
          </div>
          <fieldset>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  data-email-interval={900}
                  data-enable-email="true"
                  id="emailNotificationMinutes"
                  name="emailNotifications"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="Every {count} minutes"
                  id="user.settings.notifications.email.everyXMinutes"
                  values={
                    Object {
                      "count": 15,
                    }
                  }
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  data-email-interval={3600}
                  data-enable-email="true"
                  id="emailNotificationHour"
                  name="emailNotifications"
                  onChange={[Function]}
                  type="radio"
                />
                <FormattedMessage
                  defaultMessage="Every hour"
                  id="user.settings.notifications.email.everyHour"
                />
              </label>
            </div>
          </fieldset>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                data-email-interval={0}
                data-enable-email="false"
                id="emailNotificationNever"
                name="emailNotifications"
                onChange={[Function]}
                type="radio"
              />
              <FormattedMessage
                defaultMessage="Never"
                id="user.settings.notifications.email.never"
              />
            </label>
          </div>
          <div
            className="mt-3"
          >
            <FormattedMessage
              defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes."
              id="user.settings.notifications.emailInfo"
            />
             
            <FormattedMessage
              defaultMessage="Notifications received over the time period selected are combined and sent in a single email."
              id="user.settings.notifications.emailBatchingInfo"
            />
          </div>
        </fieldset>,
      ]
    }
    saving={false}
    section=""
    server_error=""
    submit={[Function]}
    title="Email Notifications"
    updateSection={[Function]}
  >
    <section
      className="section-max form-horizontal "
    >
      <h4
        className="col-sm-12 section-title"
        id="settingTitle"
      >
        Email Notifications
      </h4>
      <div
        className="col-sm-9 col-sm-offset-3"
      >
        <div
          className="setting-list"
          tabIndex="-1"
        >
          <div
            className="setting-list-item"
          >
            <fieldset
              key="userNotificationEmailOptions"
            >
              <legend
                className="form-legend"
              >
                <FormattedMessage
                  defaultMessage="Send email notifications"
                  id="user.settings.notifications.email.send"
                >
                  <span>
                    Send email notifications
                  </span>
                </FormattedMessage>
              </legend>
              <div
                className="radio"
              >
                <label>
                  <input
                    checked={false}
                    data-email-interval={30}
                    data-enable-email="true"
                    id="emailNotificationImmediately"
                    name="emailNotifications"
                    onChange={[Function]}
                    type="radio"
                  />
                  <FormattedMessage
                    defaultMessage="Immediately"
                    id="user.settings.notifications.email.immediately"
                  >
                    <span>
                      Immediately
                    </span>
                  </FormattedMessage>
                </label>
              </div>
              <fieldset>
                <div
                  className="radio"
                >
                  <label>
                    <input
                      checked={false}
                      data-email-interval={900}
                      data-enable-email="true"
                      id="emailNotificationMinutes"
                      name="emailNotifications"
                      onChange={[Function]}
                      type="radio"
                    />
                    <FormattedMessage
                      defaultMessage="Every {count} minutes"
                      id="user.settings.notifications.email.everyXMinutes"
                      values={
                        Object {
                          "count": 15,
                        }
                      }
                    >
                      <span>
                        Every 15 minutes
                      </span>
                    </FormattedMessage>
                  </label>
                </div>
                <div
                  className="radio"
                >
                  <label>
                    <input
                      checked={false}
                      data-email-interval={3600}
                      data-enable-email="true"
                      id="emailNotificationHour"
                      name="emailNotifications"
                      onChange={[Function]}
                      type="radio"
                    />
                    <FormattedMessage
                      defaultMessage="Every hour"
                      id="user.settings.notifications.email.everyHour"
                    >
                      <span>
                        Every hour
                      </span>
                    </FormattedMessage>
                  </label>
                </div>
              </fieldset>
              <div
                className="radio"
              >
                <label>
                  <input
                    checked={true}
                    data-email-interval={0}
                    data-enable-email="false"
                    id="emailNotificationNever"
                    name="emailNotifications"
                    onChange={[Function]}
                    type="radio"
                  />
                  <FormattedMessage
                    defaultMessage="Never"
                    id="user.settings.notifications.email.never"
                  >
                    <span>
                      Never
                    </span>
                  </FormattedMessage>
                </label>
              </div>
              <div
                className="mt-3"
              >
                <FormattedMessage
                  defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes."
                  id="user.settings.notifications.emailInfo"
                >
                  <span>
                    Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes.
                  </span>
                </FormattedMessage>
                 
                <FormattedMessage
                  defaultMessage="Notifications received over the time period selected are combined and sent in a single email."
                  id="user.settings.notifications.emailBatchingInfo"
                >
                  <span>
                    Notifications received over the time period selected are combined and sent in a single email.
                  </span>
                </FormattedMessage>
              </div>
            </fieldset>
          </div>
          <div
            className="setting-list-item"
          >
            <hr />
            <SaveButton
              btnClass="btn-primary"
              defaultMessage={
                <FormattedMessage
                  defaultMessage="Save"
                  id="save_button.save"
                />
              }
              disabled={false}
              extraClasses=""
              onClick={[Function]}
              saving={false}
              savingMessage={
                <FormattedMessage
                  defaultMessage="Saving"
                  id="save_button.saving"
                />
              }
            >
              <button
                className="save-button btn btn-primary"
                data-testid="saveSetting"
                disabled={false}
                id="saveSetting"
                onClick={[Function]}
                type="submit"
              >
                <LoadingWrapper
                  loading={false}
                  text={
                    <FormattedMessage
                      defaultMessage="Saving"
                      id="save_button.saving"
                    />
                  }
                >
                  <span>
                    <FormattedMessage
                      defaultMessage="Save"
                      id="save_button.save"
                    >
                      <span>
                        Save
                      </span>
                    </FormattedMessage>
                  </span>
                </LoadingWrapper>
              </button>
            </SaveButton>
            <button
              className="btn btn-sm btn-cancel cursor--pointer style--none"
              id="cancelSetting"
              onClick={[Function]}
            >
              <FormattedMessage
                defaultMessage="Cancel"
                id="setting_item_max.cancel"
              >
                <span>
                  Cancel
                </span>
              </FormattedMessage>
            </button>
          </div>
        </div>
      </div>
    </section>
  </SettingItemMax>
</EmailNotificationSetting>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, not send email notifications 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div
        className="pt-2"
      >
        <FormattedMessage
          defaultMessage="Email notifications have not been enabled by your System Administrator."
          id="user.settings.notifications.email.disabled_long"
        />
      </div>,
    ]
  }
  saving={false}
  section="email"
  server_error=""
  title="Email Notifications"
  updateSection={[Function]}
/>
`;

exports[`components/user_settings/notifications/EmailNotificationSetting should match snapshot, on serverError 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <fieldset>
        <legend
          className="form-legend"
        >
          <FormattedMessage
            defaultMessage="Send email notifications"
            id="user.settings.notifications.email.send"
          />
        </legend>
        <div
          className="radio"
        >
          <label>
            <input
              checked={false}
              data-email-interval={30}
              data-enable-email="true"
              id="emailNotificationImmediately"
              name="emailNotifications"
              onChange={[Function]}
              type="radio"
            />
            <FormattedMessage
              defaultMessage="Immediately"
              id="user.settings.notifications.email.immediately"
            />
          </label>
        </div>
        <div
          className="radio"
        >
          <label>
            <input
              checked={true}
              data-email-interval={0}
              data-enable-email="false"
              id="emailNotificationNever"
              name="emailNotifications"
              onChange={[Function]}
              type="radio"
            />
            <FormattedMessage
              defaultMessage="Never"
              id="user.settings.notifications.email.never"
            />
          </label>
        </div>
        <div
          className="mt-3"
        >
          <FormattedMessage
            defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away for more than 5 minutes."
            id="user.settings.notifications.emailInfo"
          />
           
        </div>
      </fieldset>,
    ]
  }
  saving={false}
  section=""
  server_error="serverError"
  submit={[Function]}
  title="Email Notifications"
  updateSection={[Function]}
/>
`;
